services/datamanager/builtin/sync/exponential_retry.go

```diff
@@ -168,6 +168,14 @@ func getNextWait(lastWait time.Duration, isOffline bool) time.Duration {
 // terminalError returns true if retrying will never succeed so that
 // the data gets moved to the corrupted data directory and false otherwise.
 func terminalError(err error) bool {
-	errStatus := status.Convert(err)
-	return errStatus.Code() == codes.InvalidArgument || errors.Is(err, proto.Error)
+	if status.Convert(err).Code() == codes.InvalidArgument || errors.Is(err, proto.Error) {
+		return true
+	}
+
+	for _, e := range terminalCaptureFileErrs {
```

Comment:
https://github.com/viamrobotics/rdk/pull/4557/files#diff-d4e29d2325615faf57196f0792e0f204e4a16c442bcc44924cfc35675321ba7aR25-R29
This ensures that we give up if the capture file is invalid.